### PR TITLE
Gc improvement

### PR DESCRIPTION
**Pull request by evilaffliction**
Improve redis lib garbage collection
1. Provide method with a custom reply data reader
2. Provide buffered methods for internal reader

### CODE REVIEW

File: commands.go [hunk 1]
```diff
@@ -3,12 +3,16 @@ package redis
 import (
 	"context"
 	"errors"
+	"github.com/go-redis/redis/v9/internal/proto"
```
Comment (markaylett) [inline commands.go]: Pedantic: should this go after "github.com/go-redis/redis/v9/internal" at the bottom of the import list?

**Review by markaylett — APPROVED** [review]

File: internal/proto/reader.go [hunk 2]
```diff
@@ -211,6 +211,26 @@ func (r *Reader) readBigInt(line []byte) (*big.Int, error) {
...
 
+func (r *Reader) readStringReplyBuffered(line []byte, buf []byte) (int, error) {
+	n, err := replyLen(line)
+	if err != nil {
+		return -1, err
+	}
+
+	if n+2 > len(buf) {
+		// TODO: worth improving error messaging to notify user about small buffer
+		return -1, errors.New("redis string buffer is too small")
```
Comment (markaylett) [inline internal/proto/reader.go]: Pedantic: other error messages start with "redis:"

**Comment by markaylett** [comment]
Can you squash all commits in this PR? We use convention commit messages. Should we do here also even though this is an external library?

File: commands.go (hunk 1 above)
Comment (evilaffliction) [reply commands.go]: Oh, that's my fault. Any settings available in attlassian for goimports?

Comment (evilaffliction) [reply commands.go]: fixed

File: internal/proto/reader.go (hunk 2 above)
Comment (evilaffliction) [reply internal/proto/reader.go]: fixed

**Comment by evilaffliction** [comment]
Sure. I do always squash commits.
Can I use this manual to create commit message?
https://reactivemarkets.atlassian.net/wiki/spaces/ENG/pages/456558542/Policy+SDLC

**Comment by evilaffliction** [comment]
moved to https://github.com/reactive-go/redis/pull/2

**Pull request closed without merging**